### ISSUE TITLE
Schema Diff tool only works first time (per window)

### ISSUE THREAD

**Issue by Bilge** [pgadmin-org/pgadmin4#7132]
**Describe the bug**

Schema Diff only works on first compare (and possibly subsequent compares, provided the source and target remain unchanged).

**To Reproduce**

1. Tools > Schema Diff
2. Compare the schema of _Database A_ against _Database B_
3. In the same window, invert the source and target to compare _Database B_ against _Database A_

**Expected behavior**

Diff is inverted.

**Error message**

No error, but the diff is just complete nonsense. A completely different (and correct) result of comparing _Database B_ against _Database A_ is produced when opening a new Schema Diff window.

Aside, there should be a button to quickly invert the source and target selections.

**Desktop (please complete the following information):**
 - OS: Windows
 - Version: 8.2
 - Mode: Desktop
 - Package type: MSI installer

**Additional context**

It doesn't seem to matter whether just the database is compared or a specific schema within the database; the bug persists either way, provided the source and/or target database changes between compares.

**Comment by anilsahoo20** [comment]
Hi @Bilge,
Can you please upgrade to pgAdmin 4 v8.2 and check if still issue persists?

**Comment by Bilge** [comment]
It does.

**Comment by akshay-joshi** [comment]
@Bilge 

I have tried with v8.2 and the issue is not reproducible. I have created two databases 'src' and 'tgt'. 

- I created a table with columns 'col1' and 'col2' in 'src'.
- I have created the same table with only one column 'col1' in 'tgt'.
- On comparing 'src' with 'tgt' it shows the 'ADD COLUMN...' query while when inverting the source and target it shows the 'DROP COLUMN..' syntax which is perfectly fine.

<img width="1314" alt="Src- Tgt" src="https://github.com/pgadmin-org/pgadmin4/assets/1632882/7ccc0277-06fe-4715-9402-8b757ec88008">


<img width="1399" alt="Tgt- Src" src="https://github.com/pgadmin-org/pgadmin4/assets/1632882/8fcb8135-f188-4230-8697-7578046b3484">

Can you please elaborate more on what went wrong in your case? Can you please share the screenshot?

**Comment by Bilge** [comment]
## Source -> Target

![image](https://github.com/pgadmin-org/pgadmin4/assets/470626/3d6a7b40-7e79-4ea3-8c15-f88a6c2af1b6)

## Target -> Source

Switch the instances only (do not touch the schema selection).

![image](https://github.com/pgadmin-org/pgadmin4/assets/470626/d8f69db2-1f81-4be2-a4e6-af72ac941580)

Suddenly all the tables are missing and it finds an extension is the difference instead.

### Clear schema and re-select

Instances are still switched, but let's try re-selecting the schemas since the database instances have been swapped.

![image](https://github.com/pgadmin-org/pgadmin4/assets/470626/754bfdd9-ce55-4c00-b409-4053897c9c9c)

Now it's even worse, there's just a giant error instead.

Bottom line: schema diff only works first time.

**Comment by akshay-joshi** [comment]
@Bilge 

Can you please share the logs related to schema diff? You can view it from "Runtime -> View log..."
If possible identify for which objects(table, function, etc...) it is throwing an error and send the DDL if possible

**Comment by akshay-joshi** [comment]
No response from the author, hence closing it.
